src/conn/conn_compact.c

```diff
@@ -18,6 +21,61 @@ __compact_server_run_chk(WT_SESSION_IMPL *session)
     return (FLD_ISSET(S2C(session)->server_flags, WT_CONN_SERVER_COMPACT));
 }
 
+/*
+ * __background_compact_find_next_uri --
+ *     Given a URI, find the next one in the metadata file that is eligible for compaction.
+ */
+static int
+__background_compact_find_next_uri(WT_SESSION_IMPL *session, WT_ITEM *uri, WT_ITEM *next_uri)
+{
+    WT_CURSOR *cursor;
+    WT_DECL_RET;
+    int exact;
+    const char *key;
+
+    cursor = NULL;
+    exact = 0;
+    key = NULL;
+
+    /* Use a metadata cursor to have access to the existing URIs. */
+    WT_ERR(__wt_metadata_cursor(session, &cursor));
+
+    /* Position the cursor on the given URI. */
+    cursor->set_key(cursor, (const char *)uri->data);
+    WT_ERR(cursor->search_near(cursor, &exact));
+
+    /*
+     * The given URI may not exist in the metadata file. Since we always want to return a URI that
+     * is lexicographically larger the given one, make sure not to go backwards.
+     */
+    if (exact <= 0)
+        WT_ERR(cursor->next(cursor));
+
+    /* Loop through the eligible candidates. */
+    while (ret == 0) {
```

Comment:
This loop conditional does not appear to serve any purpose. The loop body exits either through `WT_ERR` or `break`.  You could potentially invert:
```suggestion
    do {
    } while ((ret = cursor->next(cursor)) != ERR);
```
Or use `for (;;)`.
By no means a technical problem when scanning I am looking review I am looking for how the loop might terminate and this gave me the impression that something was missing.

Reply:
Thanks, I agree! Let me know what you think of 80ac032ba83f7d476e8eed671d6075439693deb2.